lib/opte/src/engine/snat.rs

```diff
@@ -246,37 +246,38 @@ impl<T: ConcreteIpAddr + 'static> SNat<T> {
 
         let echo_ident = match T::MESSAGE_PROTOCOL {
             Protocol::ICMP => {
-                let icmp = meta
-                    .inner_icmp()
-                    .ok_or(GenIcmpErr::<Icmpv4Message>::MetaNotFound)?;
-                if icmp.ty() != u8::from(Icmpv4Message::EchoRequest) {
-                    Err(GenIcmpErr::NotRequest(icmp.ty()).into())
+                let icmp = meta.inner_icmp().ok_or(GenIcmpErr::MetaNotFound)?;
+
+                Ok(if icmp.ty() == u8::from(Icmpv4Message::EchoRequest) {
+                    icmp.echo_id()
                 } else {
-                    Ok(icmp.echo_id())
-                }
+                    None
+                })
             }
             Protocol::ICMPv6 => {
-                let icmp6 = meta
-                    .inner_icmp6()
-                    .ok_or(GenIcmpErr::<Icmpv6Message>::MetaNotFound)?;
-                if icmp6.ty() != u8::from(Icmpv6Message::EchoRequest) {
-                    Err(GenIcmpErr::NotRequest(icmp6.ty()).into())
+                let icmp6 =
+                    meta.inner_icmp6().ok_or(GenIcmpErr::MetaNotFound)?;
+
+                Ok(if icmp6.ty() == u8::from(Icmpv6Message::EchoRequest) {
```

Comment:
Are we not handling echo reply here because this is for initializing a few flow only?

Reply:
In general for SNAT, Echo Reply messages should all be handled by the LFT entry created as part of a stateful allow here.

The other thing is that this code is only used in the outbound direction. The only time a guest should send an echo reply would be unsolicited (?), or if the other side quickly follows up with its own echo request on the same identifier (which is, again, handled by the pair of LFT entries created by the first outbound packet).